CONTRIBUTING.md

```diff
@@ -0,0 +1,122 @@
+# How to Contribute to BotKube
+
+We'd love your help!
+
+BotKube is [MIT Licensed](LICENSE) and accepts contributions via
+GitHub pull requests. This document outlines some of the conventions
+on development workflow, commit message formatting, contact points and
+other resources to make it easier to get your contributions accepted.
+
+We gratefully welcome improvements to
+[documentation](https://www.botkube.io/ "Go to documentation site") as
+well as to code.
+
+## Contributing to documentation
+You can contribute to documentation by following [these
+instructions](https://github.com/infracloudio/botkube-docs#contributing
+"Contributing to BotKube Docs")
+
+## Compile BotKube from source code
+### Prerequisite
+* Make sure you have `go` compiler installed.
+* BotKube uses [`dep`](https://github.com/golang/dep) to manage
+dependencies.
+* You will also need `make` and
+[`docker`](https://docs.docker.com/install/) installed on your
+machine.
+* Clone the source code
+   ```sh
+   $ git clone https://github.com/infracloudio/botkube.git $GOPATH/src/github.com/infracloudio/botkube
+   $ cd $GOPATH/src/github.com/infracloudio/botkube
+   ```
+
+Now you can build and run BotKube by one of the following ways
+### Build the container image
+1. This will build BotKube and create a new container image tagged as `infracloud/botkube:latest`
+   ```sh
+   $ make build
+   $ docker tag infracloud/botkube:latest <your_account>/botkube:latest
+   $ docker push <your_account>/botkube:latest
+   ```
+   Where `<your_account>` is Docker hub account to which you can push the image
+
+2. Follow the [instructions from
+   README.md](https://github.com/infracloudio/botkube#using-helm) to
+   deploy newly created image in your cluster.
+   ```sh
+   # Set the container image
+   $ helm install --name botkube --namespace botkube \
+	   ... other options ...
+	   --set image.repository=your_account/botkube \
+	   --set image.tag=latest
+	   ...
+	   helm/botkube
+   ```
+
+### Build and run BotKube locally
+1. Build BotKube binary  
+   If you don't want to build the container image, you can build the
+   binary like this,
+   ```sh
+   # Fetch the dependencies
+   $ dep ensure
+   # Build the binary
+   $ go build ./cmd/botkube/
```

Comment:
@bhavin192 Should add a rule Makefile for this? As of now, `make build` builds docker image for Botkube. Instead of that we can have `make build`- to build binary and `make image` - to build docker image.

Reply:
Yes, that should be done but I believe that can be handled as separate issue/PR, we should close on this PR for now.

Reply:
Created an issue to track that